src/gpio.rs

```diff
@@ -630,6 +473,27 @@ macro_rules! gpio {
                         unsafe { (*$GPIOX::ptr()).idr.read().bits() & (1 << $i) == 0 }
                     }
                 }
+
+                impl<MODE> $PXi<MODE> {
```

Comment:
The alternative of the surrounding `impl<MODE>` would be to have that impl in the macro itself which would be a better solution IMHO, but would also clutter the doc, because each `into_af` would be in its own impl block. Any thoughts?

Reply:
I think it is fine as is, cluttering the docs is definitely something we want to avoid.